scipy/stats/_qmc.py

```diff
@@ -0,0 +1,1500 @@
+        if not (np.all(sample >= 0) and np.all(sample <= 1)):
+            raise ValueError('Sample is not in hypercube')
+
+        return sample * (upper - lower) + lower
+    else:
+        # Checking that sample is within the bounds
+        if not (np.all(sample >= lower) and np.all(sample <= upper)):
+            raise ValueError('Sample is out of bounds')
+
+        return (sample - lower) / (upper - lower)
+
+
+def discrepancy(sample, iterative=False, method='CD'):
+    """Discrepancy on a given sample.
+
+    Parameters
+    ----------
+    sample : array_like (n, d)
+        The sample to compute the discrepancy from.
+    iterative : bool, optional
+        Must be False if not using it for updating the discrepancy.
+        Default is False. Refer to the notes for more details.
+    method : str, optional
+        Type of discrepancy, can be ['CD', 'WD', 'MD', 'star']. Refer to
+        the notes for more details. Default is ``CD``.
+
+    Returns
+    -------
+    discrepancy : float
+        Discrepancy.
+
+    Notes
+    -----
+    The discrepancy is a uniformity criterion used to assess the space filling
+    of a number of samples in a hypercube.
+    The discrepancy measures how the spread of the points deviates from a
+    uniform distribution.
+    The lower the value is, the better the coverage of the parameter space is.
+
+    A discrepancy quantifies the distance between the continuous uniform
+    distribution on a hypercube and the discrete uniform distribution on
+    :math:`n` distinct sample points. Smaller values are better. For a
+    collection of subsets of the hypercube, the discrepancy is the greatest
+    absolute difference between the fraction of sample points in one of those
+    subsets and the volume of that subset. There are different definitions of
+    discrepancy corresponding to different collections of subsets. Some
+    versions take a root mean square difference over subsets instead of
+    a maximum.
+
+    A measure of uniformity is reasonable if it satisfies the following
+    criteria [1]_:
+
+    1. It is invariant under permuting factors and/or runs.
+    2. It is invariant under rotation of the coordinates.
+    3. It can measure not only uniformity of the sample over the hypercube,
+       but also the projection uniformity of the sample over non-empty
+       subset of lower dimension hypercubes.
+    4. There is some reasonable geometric meaning.
+    5. It is easy to compute.
+    6. It satisfies the Koksma-Hlawka-like inequality.
+    7. It is consistent with other criteria in experimental design.
+
+    Four methods are available:
+
+    * ``CD``: Centered Discrepancy - subspace involves a corner of the
+      hypercube
+    * ``WD``: Wrap-around Discrepancy - subspace can wrap around bounds
+    * ``MD``: Mixture Discrepancy - mix between CD/WD covering more criteria
+    * ``star``: Star L2-discrepancy - like CD BUT variant to rotation
```

Comment:
```suggestion
    * ``L2-star``: L2-star discrepancy - like CD BUT variant to rotation
```
Let's call this "L2-star" to avoid confusion with the L-infinity star discrepancy, which is the one just called "star".